datadog/dogstatsd/aggregator.py

```diff
@@ -0,0 +1,77 @@
+import threading
+import time
+from datadog.dogstatsd.metrics import MetricAggregator, CountMetric, GaugeMetric, SetMetric
+from datadog.dogstatsd.metric_types import MetricType
+
+class Aggregator(object):
+    def __init__(self, client):
+        self.client = client
+
+        self.metrics_map = {
+            MetricType.COUNT: {},
+            MetricType.GAUGE: {},
+            MetricType.SET: {}
+        }
+        self.locks = {
+            MetricType.COUNT: threading.RLock(),
+            MetricType.GAUGE: threading.RLock(),
+            MetricType.SET: threading.RLock()
+        }
+
+        self.closed = threading.Event()
+        self.exited = threading.Event()
+
+    def start(self, flush_interval):
+        self.flush_interval = flush_interval
+        self.ticker = threading.Timer(self.flush_interval, self.tick)
+        self.ticker.start()
+
+    def tick(self):
+        while not self.closed.is_set():
+            self.send_metrics()
+            time.sleep(self.flush_interval)
```

Comment:
Do we need to sleep here if `tick` is being executed on a ticker every `flush_interval`?

Reply:
I don't think that the timer repeats every flush interval.

<img width="881" alt="image" src="https://github.com/user-attachments/assets/a551d332-0bfb-4f0c-9cb8-8471fd3d3c83">

It seems like it is just called once.

Is this a problem? Do we want a thread for every flush interval?

Reply:
Oh my bad I misunderstood how timers work. This looks fine :)

We already maintain a [flush loop for buffered metrics in](https://github.com/DataDog/datadogpy/blob/f20e449db05e55aff389289b55f9a57b536b37db/datadog/dogstatsd/base.py#L517) `base.py`

I wonder if this can be re-used or refactored to be shared with the aggregator since it's quite similar in usage.